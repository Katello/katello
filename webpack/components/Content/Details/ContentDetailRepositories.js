import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'patternfly-react';
import TableSchema from './ContentDetailRepositoryTableSchema';

const ContentDetailRepositories = ({ repositories }) => (
  <div>
    <Table.PfProvider columns={TableSchema}>
      <Table.Header />
      <Table.Body rows={repositories} rowKey="id" />
    </Table.PfProvider>
  </div>
);

ContentDetailRepositories.propTypes = {
  repositories: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default ContentDetailRepositories;
