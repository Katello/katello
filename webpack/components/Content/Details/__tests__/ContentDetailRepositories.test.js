import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
import ContentDetailRepositories from '../ContentDetailRepositories';

jest.mock('foremanReact/components/Pagination/PaginationWrapper');

const fixtures = {
  'renders with repositories': {
    repositories: [
      {
        id: 1,
        name: 'dummy name',
        product_id: 1,
        product_name: 'dummy product',
      },
    ],
  },
};

describe('Content detail repositories component', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(ContentDetailRepositories, fixtures));
});
