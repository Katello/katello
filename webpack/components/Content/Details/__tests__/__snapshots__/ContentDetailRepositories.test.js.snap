// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content detail repositories component rendering renders with repositories 1`] = `
<div>
  <TablePfProvider
    bordered={false}
    className=""
    columns={
      Array [
        Object {
          "cell": Object {
            "formatters": Array [
              [Function],
            ],
          },
          "header": Object {
            "formatters": Array [
              [Function],
            ],
            "label": "Name",
          },
          "property": "name",
        },
        Object {
          "cell": Object {
            "formatters": Array [
              [Function],
            ],
          },
          "header": Object {
            "formatters": Array [
              [Function],
            ],
            "label": "Product",
          },
          "property": "product_name",
        },
      ]
    }
    components={Object {}}
    condensed={false}
    dataTable={false}
    hover={false}
    inlineEdit={false}
    striped={false}
  >
    <Header />
    <Body
      onRow={[Function]}
      rowKey="id"
      rows={
        Array [
          Object {
            "id": 1,
            "name": "dummy name",
            "product_id": 1,
            "product_name": "dummy product",
          },
        ]
      }
    />
  </TablePfProvider>
</div>
`;
